machineconfiguration/v1alpha1/types_machineosconfig.go

```diff
@@ -0,0 +1,233 @@
+	// that the content can be applied to all architectures, or in a single arch cluster: the only architecture.
+	// +kubebuilder:validation:Enum:=arm64;amd64;ppc64le;s390x;aarch64;x86_64;noArch
+	// +kubebuilder:default:=noArch
+	// +optional
+	ContainerFileArch ContainerFileArch `json:"containerFileArch"`
+	// content is the custom content to be built
+	// +kubebuilder:validation:Required
+	Content string `json:"content"`
+}
+
+type ContainerFileArch string
+
+const (
+	// describes the arm64 architecture
+	Arm64 ContainerFileArch = "arm64"
+	// describes the amd64 architecture
+	Amd64 ContainerFileArch = "amd64"
+	// describes the ppc64le architecture
+	Ppc ContainerFileArch = "ppc64le"
+	// describes the s390x architecture
+	S390 ContainerFileArch = "s390x"
+	// describes the aarch64 architecture
+	Aarch64 ContainerFileArch = "aarch64"
+	// describes the x86_64 architecture
+	x86_64 ContainerFileArch = "x86_64"
+	// describes a containerfile that can be applied to any arch
+	noArch ContainerFileArch = "noArch"
+)
+
+// Refers to the name of a MachineConfigPool (e.g., "worker", "infra", etc.):
+type MachineConfigPoolReference struct {
+	// name of the MachineConfigPool object.
+	// +kubebuilder:validation:MaxLength:=253
+	// +kubebuilder:validation:Pattern=`^([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]{0,61}[a-zA-Z0-9])(\.([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]{0,61}[a-zA-Z0-9]))*$`
+	// +kubebuilder:validation:Required
+	Name string `json:"name"`
+}
+
+// Refers to the name of an image registry push/pull secret needed in the build process.
+type ImageSecretObjectReference struct {
+	// name is the name of the secret used to push or pull this MachineOSConfig object.
+	// +kubebuilder:validation:Required
+	Name string `json:"name"`
+}
+
+type TriggerRebuild string
+
+const (
+	// RebuildNow indicates that the user wants to rebuild their image immediately
+	RebuildNow TriggerRebuild = "RebuildNow"
+	// RebuildDelayed indicates that the user wants to rebuild their image the next time the build controller is idle.
+	// if there is a build in progress, a rebuild will be triggered the next time that build fails or succeeds. This is often used
+	// when the user changes the top level configuration for a pool's build.
+	RebuildDelayed TriggerRebuild = "RebuildDelayed"
+	// NoRebuild indicates the user does not want a manual rebuild, this is the default.
+	NoRebuild TriggerRebuild = "NoRebuild"
+)
+
+// MachineOSRebuild describes user options to manage rebuilding the image.
+type MachineOSRebuild struct {
+	// rebuildStrategy is the user provided manual rebuild method of choice
+	// +kubebuilder:default:=NoRebuild
+	// +kubebuilder:validation:Enum:=NoRebuild;RebuildNow;RebuildDelayed
+	// +kubebuilder:validation:Required
+	RebuildStrategy TriggerRebuild `json:"rebuildStrategy"`
+	// maxRetries is the amount of times we want to auto-rebuild after build failures before we deem the update unreconcilable.
```

Comment:
why would we ever give up instead of simply having longer and longer timeouts until we're only retrying every few hours?

Reply:
Is there a current methodology to do this? Or would this be handled on the controller side and here I would put  field called `timeout` or something.